lib/IRGen/GenStruct.cpp

```diff
@@ -367,6 +390,54 @@ namespace {
       (void)ClangDecl;
     }
 
+    void destroy(IRGenFunction &IGF, Address address, SILType T,
+                 bool isOutlined) const override {
+      auto *destructor = getCXXDestructor(T);
+      // If the destructor is trivial, clang will assert when we call
+      // `emitCXXDestructorCall` so, just let Swift handle this destructor.
+      if (!destructor || destructor->isTrivial()) {
+        // If we didn't find a destructor to call, bail out to the parent
+        // implementation.
+        StructTypeInfoBase<AddressOnlyClangRecordTypeInfo, FixedTypeInfo,
+                           ClangFieldInfo>::destroy(IGF, address, T,
+                                                    isOutlined);
+        return;
+      }
+
+      if (!destructor->isUserProvided()) {
+        assert(!destructor->isDeleted() &&
```

Comment:
I just started looking into this, but can't it happen that the user feeds us a C++ struct with deleted destructor? In that case we shouldn't crash, but show a diagnostic. Could we add a test for this?

Reply:
I think we should simply not import those types. I think in a very early version of this patch, we had logic to skip these in `ImportDecl` but that was removed with the other logic in that file. Anyway, #34649 adds it back.

Reply:
Marking as resolved because that PR landed.